### ISSUE TITLE
please, update to tf 1.0

### ISSUE THREAD

**Issue by avr248** [allenai/bi-att-flow#10]

**Comment by seominjoon** [comment]
We will update to tf 1.0 soon, thanks!

**Comment by avr248** [comment]
very thanks!

**Comment by rubby33** [comment]
Wooo!  Look forward to it!

**Comment by it99ts** [comment]
Hi - do you have a timeline for 1.0 version?

**Comment by seominjoon** [comment]
@it99ts Upgrading is not hard (need to change a few lines of code) but it has been reported that the performance drops significantly (>5%) when trained on TF 1.0 after upgrade. It seems somethings have changed internally from 0.11 to 1.0, so I will need to find this out and resolve it to upgrade to 1.0. SoI cannot give a good estimate at this point. Sorry about that!

**Comment by sathishreddy** [comment]
Hi -- I trained the model again after upgrading the model to Tensorflow 1.1. I also changed the default learning rate from 0.5 to 0.001 (model was diverging with 0.5). I was able to achieve EM: 66.89, F1: 76.65 on dev set. Thanks.

**Comment by fcjy** [comment]
@sathishreddy hi, can you push your code to github? I try to upgrading the model to Tensorflow 1.1, but meet some problems.

**Comment by sathishreddy** [comment]
@fcjy ... sure I will do that in couple of days ... will let you know once it is done. thanks.

**Comment by fcjy** [comment]
@sathishreddy thank you!

**Comment by sathishreddy** [comment]
@fcjy ... I pushed my code to github. Please checkout form [here](https://github.com/sathishreddy/bi-att-flow). Currenlty I upgraded code for SQuAD mode (code in basic, my/tensorflow directories) only. Tested in Single GPU setting. Tensorflow version is  **1.1.0**. 

@seominjoon ... Please let me know how to contribute to the main branch. I can push the code, if you are okay with it (After testing other scenarios also).  
Thanks.

**Comment by fcjy** [comment]
@sathishreddy thank you again！

**Comment by seominjoon** [comment]
@sathishreddy Great to hear, and yes, pull request would be greatly appreciated, thank you!
I guess something regarding the AdaDelta optimizer has changed internally in the new version..

**Comment by insikk** [comment]
@sathishreddy Thank you for your contribution. What GPU did you used for training? How long did it take to achieve EM: 66.89, F1: 76.65 on dev set with learning rate 0.001? I was trying with your code, but it was so slow. It seems training is not going well. Does it take very long time?

**Comment by sathishreddy** [comment]
Hi @insikk ... I used Tesla K40 gpu card... It took me approx. 2 days to train the model with lr=0.001, num epochs=12.  Each step roughly (with batch size=30) took me around 4.5 sec. Thank you.

**Comment by insikk** [comment]
According to [this patch](https://github.com/tensorflow/tensorflow/commit/c1bf10f9c62d3c62cf7aed95aab248dd0ead2cf1), adadelta optimizer had a bug, and patched since tf1.0. Do you think this may be possible cause of the significant performance drop? I may have to compare result of pre-TF1.0 and after-TF1.0 performance with other optimizer. 

@seominjoon Do you have any specific reason for choosing AdaDelta? If you did some experiments with other optimizer such as Adam, or any other ones, could you share your experience in terms of performance?

**Comment by seominjoon** [comment]
@insikk Thanks for the pointer, it now makes sense! In fact, Adam with init_lr 0.001 works well in tf v1 (thanks to Jannis jbulian@google.com). I just committed the tf v1.1 compatible version to dev branch. I will pull request to master soon.

**Comment by avr248** [comment]
thank you!
